DxDispatch/src/DirectMLHelpers/OperatorFieldTypes_generated.h

```diff
@@ -0,0 +1,1323 @@
+  static const ScalarVariant enum_value = ScalarVariant_Int32;
+};
+
+template<> struct ScalarVariantTraits<dml::ir::operatorFieldTypes::UInt32> {
+  static const ScalarVariant enum_value = ScalarVariant_UInt32;
+};
+
+template<> struct ScalarVariantTraits<dml::ir::operatorFieldTypes::Int64> {
+  static const ScalarVariant enum_value = ScalarVariant_Int64;
+};
+
+template<> struct ScalarVariantTraits<dml::ir::operatorFieldTypes::UInt64> {
+  static const ScalarVariant enum_value = ScalarVariant_UInt64;
+};
+
+template<> struct ScalarVariantTraits<dml::ir::operatorFieldTypes::Float32> {
+  static const ScalarVariant enum_value = ScalarVariant_Float32;
+};
+
+template<> struct ScalarVariantTraits<dml::ir::operatorFieldTypes::Float64> {
+  static const ScalarVariant enum_value = ScalarVariant_Float64;
+};
+
+bool VerifyScalarVariant(::flatbuffers::Verifier &verifier, const void *obj, ScalarVariant type);
+bool VerifyScalarVariantVector(::flatbuffers::Verifier &verifier, const ::flatbuffers::Vector<::flatbuffers::Offset<void>> *values, const ::flatbuffers::Vector<uint8_t> *types);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(1) UInt8 FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint8_t data_;
+
+ public:
+  UInt8()
+      : data_(0) {
+  }
+  UInt8(uint8_t _data)
+      : data_(::flatbuffers::EndianScalar(_data)) {
+  }
+  uint8_t data() const {
+    return ::flatbuffers::EndianScalar(data_);
+  }
+  void mutate_data(uint8_t _data) {
+    ::flatbuffers::WriteScalar(&data_, _data);
+  }
+};
+FLATBUFFERS_STRUCT_END(UInt8, 1);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(2) UInt16 FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint16_t data_;
+
+ public:
+  UInt16()
+      : data_(0) {
+  }
+  UInt16(uint16_t _data)
+      : data_(::flatbuffers::EndianScalar(_data)) {
+  }
+  uint16_t data() const {
+    return ::flatbuffers::EndianScalar(data_);
+  }
+  void mutate_data(uint16_t _data) {
+    ::flatbuffers::WriteScalar(&data_, _data);
+  }
+};
+FLATBUFFERS_STRUCT_END(UInt16, 2);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) UInt32 FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint32_t data_;
+
+ public:
+  UInt32()
+      : data_(0) {
+  }
+  UInt32(uint32_t _data)
+      : data_(::flatbuffers::EndianScalar(_data)) {
+  }
+  uint32_t data() const {
+    return ::flatbuffers::EndianScalar(data_);
+  }
+  void mutate_data(uint32_t _data) {
+    ::flatbuffers::WriteScalar(&data_, _data);
+  }
+};
+FLATBUFFERS_STRUCT_END(UInt32, 4);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(8) UInt64 FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint64_t data_;
+
+ public:
+  UInt64()
+      : data_(0) {
+  }
+  UInt64(uint64_t _data)
+      : data_(::flatbuffers::EndianScalar(_data)) {
+  }
+  uint64_t data() const {
+    return ::flatbuffers::EndianScalar(data_);
+  }
+  void mutate_data(uint64_t _data) {
+    ::flatbuffers::WriteScalar(&data_, _data);
+  }
+};
+FLATBUFFERS_STRUCT_END(UInt64, 8);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(1) Int8 FLATBUFFERS_FINAL_CLASS {
```

Comment:
>Int8 [](http://example.com/codeflow?start=39&length=4)

Why do we need special types? Can't we just use a template function specialization for each type 

WriteScalar and EndianScalar